test/rule_test_support.dart

```diff
@@ -254,6 +267,17 @@ class PubPackageResolutionTest extends _ContextResolutionTest {
   @mustCallSuper
   void setUp() {
     super.setUp();
+    // Check for any needlessly enabled experiments.
+    for (var experiment in experiments) {
+      var feature = ExperimentStatus.knownFeatures[experiment];
+      var releaseVersion = feature?.releaseVersion;
+      if (releaseVersion != null && releaseVersion >= minSdkVersion) {
```

Comment:
I'm not sure why we need to compare the `releaseVersion` with the `minSdkVersion`. I would have expected that it's sufficient to ask whether the feature is enabled by default. What am I missing?

Reply:
I may be overthinking it but what I'm trying to guard against is the case where someone is testing locally with an unreleased SDK that enables a feature that is *not* yet enabled by default in a published SDK.  In that case, there would be no way to get the test to pass locally and not have the CI break.

Reply:
Check my reasoning.

When we're running a test, the code under test is being analyzed by the version of the `analyzer` package that was selected by pub (usually that latest published version). That same version of the `analyzer` package is the one being queried to see whether the feature has been enabled by default. As a result, asking the feature should be all you need to do because it's the same test that will be used by the analyzer when analyzing the code under test to find the default features.

In other words, I don't think the SDK running the test ever has any bearing on the state of enablement of the features.

Reply:
As discussed.  I agree!  Thanks for the sanity check. 🙏